rotkehlchen/chain/evm/decoding/spark/savings/decoder.py

```diff
@@ -0,0 +1,321 @@
+                else:  # swap exact out
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                    event.notes = f'Remove {amount_out} {asset_out.symbol} from Spark Savings'
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_spark_tokens_deposit_withdrawal(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes deposit and withdrawal events for Spark tokens (e.g., sUSDC).
+        Similar to PSM deposits/withdrawals but for direct token interactions.
+        """
+        # Check if this is a sky migration event on Ethereum and delegate to the migration decoder
+        if (
+            self.evm_inquirer.chain_id == ChainID.ETHEREUM and
+            context.tx_log.address == SUSDS_CONTRACT and
+            context.tx_log.topics[0] in (DEPOSIT_TOPIC, WITHDRAW_TOPIC) and
+            bytes_to_address(context.tx_log.topics[1]) == MIGRATION_ACTIONS_CONTRACT
+        ):
+            return self._decode_sky_migration_to_susds(context)
+
+        # skip if there are later deposit/withdraw events (we only want to process the last one)
+        if (
+            context.tx_log.topics[0] not in (DEPOSIT_TOPIC, WITHDRAW_TOPIC) or
+            any(
+                log.topics[0] in (DEPOSIT_TOPIC, WITHDRAW_TOPIC) and
+                log.log_index > context.tx_log.log_index
+                for log in context.all_logs
+            )
+        ):
+            return DEFAULT_DECODING_OUTPUT
+
+        vault_token = self.base.get_or_create_evm_token(
+            address=context.tx_log.address,
+            encounter=(encounter := TokenEncounterInfo(should_notify=False)),
+        )
+        underlying_token = self.base.get_or_create_evm_token(
+            address=deserialize_evm_address(self.evm_inquirer.call_contract(
+                contract_address=vault_token.evm_address,
+                abi=UMAMI_ASSET_VAULT_ABI,
+                method_name='asset',
+            )),
+            encounter=encounter,
+        )
+
+        if context.tx_log.topics[0] == DEPOSIT_TOPIC:
+            underlying_amount = token_normalized_value_decimals(
+                token_amount=int.from_bytes(context.tx_log.data[:32]),
+                token_decimals=underlying_token.decimals,
+            )
+            shares_amount = token_normalized_value_decimals(
+                token_amount=int.from_bytes(context.tx_log.data[32:64]),
+                token_decimals=vault_token.decimals,
+            )
+        else:  # WITHDRAW_TOPIC
+            shares_amount = token_normalized_value_decimals(
+                token_amount=int.from_bytes(context.tx_log.data[32:64]),
+                token_decimals=vault_token.decimals,
+            )
+            underlying_amount = token_normalized_value_decimals(
+                token_amount=int.from_bytes(context.tx_log.data[:32]),
+                token_decimals=underlying_token.decimals,
+            )
```

Comment:
this is wrong unless I'm being extremly stupid

deposit:

``` 
underlying_amount -> context.tx_log.data[:32] -> underlying_token.decimals
``` 

withdrawal
```
underlying_amount -> context.tx_log.data[:32] -> underlying_token.decimals
``` 

seems like this if is not needed and if it is needed a test is missing